x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/date/DayNameTests.java

```diff
@@ -46,48 +46,61 @@ public DayNameTests(@Name("TestCase") Supplier<TestCaseSupplier.TestCase> testCa
     @ParametersFactory
     public static Iterable<Object[]> parameters() {
         List<TestCaseSupplier> suppliers = new ArrayList<>();
-        suppliers.addAll(generateTest("2019-03-11T00:00:00.00Z", "Monday"));
-        suppliers.addAll(generateTest("2022-07-26T23:59:59.99Z", "Tuesday"));
-        suppliers.addAll(generateTest("2017-10-11T23:12:32.12Z", "Wednesday"));
-        suppliers.addAll(generateTest("2023-01-05T07:39:01.28Z", "Thursday"));
-        suppliers.addAll(generateTest("2023-02-17T10:25:33.38Z", "Friday"));
-        suppliers.addAll(generateTest("2013-06-15T22:55:33.82Z", "Saturday"));
-        suppliers.addAll(generateTest("2024-08-18T01:01:29.49Z", "Sunday"));
+
+        // UTC, English
+        suppliers.addAll(generateTest("2019-03-11T00:00:00.00Z", "Monday", "Z", "en"));
+        suppliers.addAll(generateTest("2022-07-26T23:59:59.99Z", "Tuesday", "Z", "en"));
+        suppliers.addAll(generateTest("2017-10-11T23:12:32.12Z", "Wednesday", "Z", "en"));
+        suppliers.addAll(generateTest("2023-01-05T07:39:01.28Z", "Thursday", "Z", "en"));
+        suppliers.addAll(generateTest("2023-02-17T10:25:33.38Z", "Friday", "Z", "en"));
+        suppliers.addAll(generateTest("2013-06-15T22:55:33.82Z", "Saturday", "Z", "en"));
+        suppliers.addAll(generateTest("2024-08-18T01:01:29.49Z", "Sunday", "Z", "en"));
+
+        // Other timezones and locales
+        suppliers.addAll(generateTest("2019-03-11T22:00:00.00Z", "Tuesday", "+05:00", "en"));
+        suppliers.addAll(generateTest("2019-03-11T00:00:00.00Z", "Sunday", "America/New_York", "en"));
+        suppliers.addAll(generateTest("2019-03-11T00:00:00.00Z", "lunes", "Z", "es"));
+        suppliers.addAll(generateTest("2019-03-11T00:00:00.00Z", "domingo", "America/New_York", "es"));
 
         suppliers.add(
             new TestCaseSupplier(
+                "Null",
                 List.of(DataType.DATETIME),
                 () -> new TestCaseSupplier.TestCase(
                     List.of(new TestCaseSupplier.TypedData(null, DataType.DATETIME, "date")),
-                    Matchers.startsWith("DayNameMillisEvaluator[val=Attribute[channel=0], zoneId=Z, locale=en_US]"),
+                    Matchers.startsWith("DayNameMillisEvaluator[val=Attribute[channel=0], zoneId="),
                     DataType.KEYWORD,
                     equalTo(null)
-                ).withStaticConfiguration()
+                )
```

Comment:
Should we have a withConfiguration here?

Reply:
Perhaps meaningless for `null` values?

Reply:
Exactly. A null will be null in whatever timezone, so better keep it randomized